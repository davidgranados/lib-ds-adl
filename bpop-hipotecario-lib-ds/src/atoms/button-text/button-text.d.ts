import { DefaultButtonProps } from "../../types/button";
import { ButtonBaseProps } from "../button-base";
import { HTMLPropsFor } from "../../types";

export type ButtonTextSize = "sm" | "md" | "lg";

export interface ButtonTextProps extends DefaultButtonProps, ButtonBaseProps, HTMLPropsFor<"button"> {
  size?: ButtonTextSize;
}
