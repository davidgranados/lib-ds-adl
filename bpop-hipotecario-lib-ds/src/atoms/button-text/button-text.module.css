.default {
  border: none;
  font-family: var(--font-family-sans-inter);
  font-weight: var(--font-weight-button);
}

.primary {
  background-color: var(--background-color-button-text-primary);
  color: var(--font-color-button-text-primary-enabled);
}

.primary:focus {
  text-decoration: underline;
}

.primary:disabled {
  color: var(--font-color-button-text-primary-disabled);
}

.secondary {
  background-color: var(--background-color-button-text-secondary);
  color: var(--font-color-button-text-secondary-enabled);
}

.secondary:focus {
  text-decoration: underline;
}

.secondary:disabled {
  color: var(--font-color-button-text-secondary-disabled);
}

.size-sm {
  font-size: var(--font-size-button-text-sm);
}

.size-md {
  font-size: var(--font-size-button-text-md);
}

.size-lg {
  font-size: var(--font-size-button-text-lg);
}
