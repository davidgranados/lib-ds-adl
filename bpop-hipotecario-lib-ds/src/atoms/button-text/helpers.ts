import { ButtonTextSize, ButtonVariant } from "../../types";

export const validVariants: ButtonVariant[] = ["primary", "secondary"];
export const validSizes: ButtonTextSize[] = ["sm", "md", "lg"];
