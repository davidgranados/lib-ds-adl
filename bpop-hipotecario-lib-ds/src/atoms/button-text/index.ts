export { default as ButtonText } from "./button-text";

export { validSizes, validVariants } from "./helpers";

export type { ButtonTextProps, ButtonTextSize } from "./button-text.d";
