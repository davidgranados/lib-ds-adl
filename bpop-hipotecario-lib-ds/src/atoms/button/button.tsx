import React from "react";
import classNames from "classnames";

import { ButtonProps } from "../../types";

import buttonStyles from "./button.module.css";

export const Button: React.FC<ButtonProps> = ({
  label,
  type = "button",
  variant = "primary",
  size = "md",
  ...props
}) => {
  const buttonClassName = classNames(buttonStyles["default"], buttonStyles[variant], buttonStyles[`size-${size}`]);
  return (
    <button type={type} className={buttonClassName} {...props}>
      {label}
    </button>
  );
};
