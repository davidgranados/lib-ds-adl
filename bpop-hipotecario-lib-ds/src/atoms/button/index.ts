export { default as Button } from "./button";

export { validSizes, validVariants } from "./helpers";
