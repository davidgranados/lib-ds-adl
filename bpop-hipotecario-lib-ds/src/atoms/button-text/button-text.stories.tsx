/* istanbul ignore file */
import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import ButtonText from "./button-text";
import { validSizes, validVariants } from "./helpers";

export default {
  title: "Atoms/ButtonText",
  component: ButtonText,
  args: {
    variant: "primary",
    disabled: false,
    text: "Button Text",
  },
  argTypes: {
    variant: { options: validVariants, control: "select" },
    size: { options: validSizes, control: "select" },
    disabled: { control: "boolean" },
    onFocus: { action: "focused" },
    onBlur: { action: "blurred" },
    onClick: { action: "clicked" },
  },
} as ComponentMeta<typeof ButtonText>;

const Template: ComponentStory<typeof ButtonText> = (args) => <ButtonText {...args} />;

export const Default = Template.bind({});
