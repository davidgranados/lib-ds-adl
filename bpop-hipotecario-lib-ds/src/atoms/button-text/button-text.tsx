import classNames from "classnames";
import React from "react";

import { ButtonTextProps } from "../../types/button";

import buttonTextStyles from "./button-text.module.css";

export const ButtonText: React.FC<ButtonTextProps> = ({
  label,
  type = "button",
  variant = "primary",
  size = "md",
  ...props
}) => {
  const buttonTextClassName = classNames(
    buttonTextStyles["default"],
    buttonTextStyles[variant],
    buttonTextStyles[`size-${size}`]
  );
  return (
    <button type={type} className={buttonTextClassName} {...props}>
      {label}
    </button>
  );
};
