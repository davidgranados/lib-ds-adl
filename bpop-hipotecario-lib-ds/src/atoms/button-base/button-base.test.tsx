import React from "react";
import { mount, ReactWrapper, shallow } from "enzyme";

import { ButtonTextSize } from "../button-text";
import { ButtonSize } from "../button";

import ButtonBase from "./button-base";

describe("<ButtonBase />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow<typeof ButtonBase>(<ButtonBase>{"button"}</ButtonBase>);
    expect(wrapper).toMatchSnapshot();
  });
});
describe("<ButtonBase /> rendered in HOC", () => {
  let wrapper: ReactWrapper<typeof ButtonBase>;
  const size: ButtonTextSize | ButtonSize = "md";
  const handleFocus = jest.fn();
  const handleBlur = jest.fn();
  const onClick = jest.fn();
  const handleMouseEnter = jest.fn();
  const handleMouseLeave = jest.fn();
  const handleMouseDown = jest.fn();
  const handleMouseUp = jest.fn();
  const handleTouchStart = jest.fn();
  const handleTouchEnd = jest.fn();
  describe(" -> <ButtonText />", () => {
    beforeEach(() => {
      jest.clearAllMocks();
      // given
      wrapper = mount<typeof ButtonBase>(
        <ButtonBase
          fontSize={size}
          className={"buttonTextClassname"}
          onFocus={handleFocus}
          onBlur={handleBlur}
          onClick={onClick}
        >
          {"button-text"}
        </ButtonBase>
      );
    });
    it("should match snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("should handle focus", () => {
      // when
      wrapper.find("button").simulate("focus");
      // then
      expect(handleFocus).toBeCalled();
    });

    it("should handle blur", () => {
      // when
      wrapper.find("button").simulate("blur");
      // then
      expect(handleBlur).toBeCalled();
    });

    it("should handle click", () => {
      // when
      wrapper.find("button").simulate("click");
      // then
      expect(onClick).toBeCalled();
    });
  });
  describe(" -> <Button />", () => {
    beforeEach(() => {
      jest.clearAllMocks();
      // given
      wrapper = mount<typeof ButtonBase>(
        <ButtonBase
          fontSize={size}
          className={"buttonClassname"}
          onMouseEnter={handleMouseEnter}
          onMouseLeave={handleMouseLeave}
          onMouseDown={handleMouseDown}
          onMouseUp={handleMouseUp}
          onTouchStart={handleTouchStart}
          onTouchEnd={handleTouchEnd}
          onFocus={handleFocus}
          onBlur={handleBlur}
          onClick={onClick}
        >
          {"button"}
        </ButtonBase>
      );
    });
    it("should match snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("should handle hover", () => {
      // when
      wrapper.find("button").simulate("mouseenter");
      wrapper.find("button").simulate("mouseleave");
      // then
      expect(handleMouseEnter).toBeCalled();
      expect(handleMouseLeave).toBeCalled();
    });

    it("should handle mouse down/up events", () => {
      // when
      wrapper.find("button").simulate("mousedown");
      wrapper.find("button").simulate("mouseup");
      // then
      expect(handleMouseDown).toBeCalled();
      expect(handleMouseUp).toBeCalled();
    });

    it("should handle touch start/end events", () => {
      // when
      wrapper.find("button").simulate("touchstart");
      wrapper.find("button").simulate("touchend");
      // then
      expect(handleTouchStart).toBeCalled();
      expect(handleTouchEnd).toBeCalled();
    });

    it("should handle focus", () => {
      // when
      wrapper.find("button").simulate("focus");
      // then
      expect(handleFocus).toBeCalled();
    });

    it("should handle blur", () => {
      // when
      wrapper.find("button").simulate("blur");
      // then
      expect(handleBlur).toBeCalled();
    });

    it("should handle click", () => {
      // when
      wrapper.find("button").simulate("click");
      // then
      expect(onClick).toBeCalled();
    });
  });
});
