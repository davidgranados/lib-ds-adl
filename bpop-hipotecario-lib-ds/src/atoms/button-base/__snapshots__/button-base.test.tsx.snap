// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonBase /> rendered in HOC <ButtonBase /> rendered in <Button /> should match snapshot 1`] = `
<button
  className="default font-size-md buttonClassname"
  onBlur={[MockFunction]}
  onClick={[MockFunction]}
  onFocus={[MockFunction]}
  onMouseDown={[MockFunction]}
  onMouseEnter={[MockFunction]}
  onMouseLeave={[MockFunction]}
  onMouseUp={[MockFunction]}
  onTouchEnd={[MockFunction]}
  onTouchStart={[MockFunction]}
  type="button"
>
  button
</button>
`;

exports[`<ButtonBase /> rendered in HOC <ButtonBase /> rendered in <ButtonText /> should match snapshot 1`] = `
<button
  className="default font-size-md buttonTextClassname"
  onBlur={[MockFunction]}
  onClick={[MockFunction]}
  onFocus={[MockFunction]}
  type="button"
>
  button-text
</button>
`;

exports[`<ButtonBase /> should match snapshot 1`] = `
<button
  className="default font-size-md"
  type="button"
>
  button
</button>
`;
